### PR TITLE
Added seccomp profile support

### PR DESCRIPTION
**Pull request by kravivar**
@matthewdfuller : Please review

CC: @anshupande 

Fixes : https://github.com/adobe-community/issues-ethos/issues/1645

### CODE REVIEW

**Comment by kravivar** [comment]
Fixes : https://github.com/adobe-community/issues-ethos/issues/2546

**Comment by matthewdfuller** [comment]
@kravivar - can you hardcode the JSON into the script? Reason being is that the script is loaded into the Mesos execution environment [here](https://git.corp.adobe.com/adobe-platform/dcos-universe-private/blob/ethos-1.x-unstable/repo/packages/A/aqua-ethos-setup/0/marathon.json.mustache#L6). Since it's pretty short, we can get away with just using `EOF` and not having to download two files.

**Comment by matthewdfuller** [comment]
Also, once I merge this in, can you submit an update for the above package?

**Comment by kravivar** [comment]
@matthewdfuller : Updated the contents as per recommendation

File: aqua-ethos-setup/aqua_ethos.sh [hunk 1]
```diff
@@ -37,6 +37,30 @@ if [[ -z $TOKEN ]]; then
 	exit 1
 fi
 
+# Add the secomp profile if it doesnt exist
+read -r -d '' _SECCOMP_PROFILE <<'EOF'
+{"defaultAction":"SCMP_ACT_ERRNO","syscalls":[{"names":["capget","capset","chdir","fchown","futex","getdents64","getpid","getppid","lstat","openat","prctl","setgid","setgroups","setuid","stat","rt_sigaction","mprotect","brk","close","open","write","mmap","rt_sigprocmask","sched_getaffinity","arch_prctl","access","getrandom","sigaltstack","getrlimit","set_tid_address","fstat","stat","setsockopt","read","openat","clone","set_robust_list","ioctl","execve","gettid","socket","munmap","futex","bind"],"action":"SCMP_ACT_ALLOW","args":[],"comment":"Necessary syscalls for working of container","includes":{},"excludes":{}}]}
```
Comment (matthewdfuller) [inline aqua-ethos-setup/aqua_ethos.sh]: Sorry to be nitpicky on this, but can you use the same spacing/line breaks as you do below? It makes it very hard to read/update in the future as one big line.

**Review by matthewdfuller — CHANGES_REQUESTED** [review]
Please include this rule check in the loop at the end as well:

https://github.com/kravivar/dcos-scripts/blob/776e4bf64ef955e71fd22f1dabd2ff678ecc9507/aqua-ethos-setup/aqua_ethos.sh#L100

I would rename `EXISTING_RULE` for your addition to `EXISTING_SECOMP_RULE` or something similar

**Comment by kravivar** [comment]
@matthewdfuller I have made the changes you wanted, please review.

File: aqua-ethos-setup/aqua_ethos.sh [hunk 2]
```diff
@@ -37,10 +37,44 @@ if [[ -z $TOKEN ]]; then
...
+EXISTING_SECOMP_RULE=$(curl --write-out %{http_code} --silent --output /dev/null -H "$HEADER: Bearer $TOKEN" $WEB_URL/securityprofiles/seccomp-default)
+if [[ "$EXISTING_SECOMP_RULE" == "200" ]]; then
+        log "seccomp-default exists..."
+else
+        curl --silent -H "Content-Type: application/json" -H "$HEADER: Bearer $TOKEN" -X POST -d "$_POST_VAR" $WEB_URL/securityprofiles
+fi
+
 # See if profile already exists
-EXISTING_RULE=$(curl --write-out %{http_code} --silent --output /dev/null -H "$HEADER: Bearer $TOKEN" $WEB_URL/adminrules/core-user-rule)
+EXISTING_SECOMP_RULE=$(curl --write-out %{http_code} --silent --output /dev/null -H "$HEADER: Bearer $TOKEN" $WEB_URL/adminrules/core-user-rule)
```
Comment (matthewdfuller) [inline aqua-ethos-setup/aqua_ethos.sh]: @kravivar - why are you overwriting the existing rule `EXISTING_RULE` with `EXISTING_SECOMP_RULE` when `EXISTING_SECOMP_RULE` already exists above (see line 67)?

File: aqua-ethos-setup/aqua_ethos.sh [hunk 3]
```diff
@@ -37,10 +37,44 @@ if [[ -z $TOKEN ]]; then
...
 
-if [[ "$EXISTING_RULE" == "200" ]]; then
+if [[ "$EXISTING_SECOMP_RULE" == "200" ]]; then
```
Comment (matthewdfuller) [inline aqua-ethos-setup/aqua_ethos.sh]: This should go back to `EXISTING_RULE`

File: aqua-ethos-setup/aqua_ethos.sh [hunk 4]
```diff
@@ -73,23 +107,23 @@ curl --silent -H "$HEADER: Bearer $TOKEN" -X PUT -d @$CRED_DIR/configs/threat_mi
 
 sudo rm -rf $CRED_DIR/configs
 
-while [[ "$EXISTING_PROFILE" == "200" && "$EXISTING_RULE" == "200" ]]; do
+while [[ "$EXISTING_PROFILE" == "200" && "$EXISTING_SECOMP_RULE" == "200" ]]; do
```
Comment (matthewdfuller) [inline aqua-ethos-setup/aqua_ethos.sh]: This should become:
```
while [[ "$EXISTING_PROFILE" == "200" && "$EXISTING_PROFILE" == "200" && "$EXISTING_SECOMP_RULE" == "200" ]]; do
```

File: aqua-ethos-setup/aqua_ethos.sh [hunk 5]
```diff
@@ -73,23 +107,23 @@ curl --silent -H "$HEADER: Bearer $TOKEN" -X PUT -d @$CRED_DIR/configs/threat_mi
...
 	log "Profile and rule are still active..."
 
 	if [[ $(expr $(date +%s) - $(date +%s -r $CRED_DIR/login)) -gt 1800 ]]; then
 		login
 	fi
 
-	EXISTING_RULE=$(curl --write-out %{http_code} --silent --output /dev/null -H "$HEADER: Bearer $TOKEN" $WEB_URL/adminrules/core-user-rule)
+	EXISTING_SECOMP_RULE=$(curl --write-out %{http_code} --silent --output /dev/null -H "$HEADER: Bearer $TOKEN" $WEB_URL/adminrules/core-user-rule)
```
Comment (matthewdfuller) [inline aqua-ethos-setup/aqua_ethos.sh]: This should be an addition not a change

File: aqua-ethos-setup/aqua_ethos.sh [hunk 6]
```diff
@@ -73,23 +107,23 @@ curl --silent -H "$HEADER: Bearer $TOKEN" -X PUT -d @$CRED_DIR/configs/threat_mi
...
 	EXISTING_PROFILE=$(curl --write-out %{http_code} --silent --output /dev/null -H "$HEADER: Bearer $TOKEN" $WEB_URL/securityprofiles/Ethos)
 	
-	if [[ "$EXISTING_RULE" == "200" && "$EXISTING_PROFILE" == "200" ]]; then
+	if [[ "$EXISTING_SECOMP_RULE" == "200" && "$EXISTING_PROFILE" == "200" ]]; then
```
Comment (matthewdfuller) [inline aqua-ethos-setup/aqua_ethos.sh]: This should be:
```
if [[ "$EXISTING_RULE" == "200" && "$EXISTING_SECOMP_RULE" == "200" && "$EXISTING_PROFILE" == "200" ]]; then
```

**Review by matthewdfuller — CHANGES_REQUESTED** [review]
The recent changes broke existing work. We don't want to _replace_ `EXISTING_RULE`, we want to add a _new_ item for `EXISTING_SECOMP_RULE`. Please make these changes so we are testing both.

**Comment by matthewdfuller** [comment]
@kravivar - ping on these updates

**Pull request closed without merging**